tests/migration/prepare_addons_for_migration.pm

```diff
@@ -0,0 +1,27 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Make sure all addons are registered then remove -release packages from dropped modules before installation
+
+# Maintainer: QA SLE YaST team <qa-sle-yast@suse.de>
+
+use base 'y2_installbase';
+use strict;
+use warnings;
+use testapi;
+use migration 'remove_dropped_module_packages';
+use registration 'register_addons_cmd';
+
+
+sub run {
+    unless (script_run("SUSEConnect --status-text |grep -a3 basesystem |grep 'Not Registered'")) {
```

Comment:
- `unless` is only used in certain programming languages with the premise the condition is easy to read, in general some function or variable, but this piped command is not the case.

- this is to check if something is not register first then register it, I guess, we should use some lib function for it, is the only way to check for the basesystem or is there some more generic way?

Reply:
see previous comment, this was more of an experiment, I then wrote it better in a lib, but anyway it fails because of scc, so I dropped that altogether, the only solution is to make sure we don't launch the test after 4pm (more or less).